src/common/multi_file_reader.cpp

```diff
@@ -235,16 +236,17 @@ void MultiFileReader::BindOptions(MultiFileReaderOptions &options, MultiFileList
 }
 
 void MultiFileReader::FinalizeBind(const MultiFileReaderOptions &file_options, const MultiFileReaderBindData &options,
-                                   const string &filename, const vector<string> &local_names,
-                                   const vector<LogicalType> &global_types, const vector<string> &global_names,
+                                   const string &filename, const vector<MultiFileReaderColumnDefinition> &local_columns,
```

Comment:
Wondering now if we should perhaps use a single MultiFileReaderColumnDefinition with a top level struct type column instead of a vector. I don't have a strong opinion on this though